src/controllers/users.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const { UserModel } = require('../models/users');
+
+const getAllUsers = (req, res) => {
+  const users = UserModel.getUsers();
+
+  res.send(users);
+};
+
+const addNewUser = (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.status(400).send('Bad request');
+
+    return;
+  }
+
+  const newUser = UserModel.addUser({ name });
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+const getOneUser = (req, res) => {
+  const { userId } = req.params;
+  const user = UserModel.getUser(+userId);
+
+  if (!user) {
+    res.status(404).send('User not found');
+
+    return;
+  }
+
+  res.send(user);
+};
+
+const deleteUser = (req, res) => {
+  const { userId } = req.params;
+
+  const user = UserModel.removeUser(+userId);
+
+  if (!user) {
+    res.status(404).send('User not found');
+
+    return;
+  }
+
+  res.status(204).send('No Content');
+};
+
+const updateUser = (req, res) => {
+  const { userId } = req.params;
+  const { name } = req.body;
+
+  const updatedUser = UserModel.updateUser({
+    id: userId,
+    name,
+  });
+
+  if (!updatedUser) {
+    res.status(404).send('User not found');
+
+    return;
+  }
+
+  if (typeof name !== 'string') {
```

Comment:
The type check for `name` should be done before attempting to update the user to prevent unnecessary operations. Consider moving this check before line 57.